### PR TITLE
use mt19937 rng for BDP thermostat

### PR DESCRIPTION
**Pull request by brucefan1983**
The `ran1()` function in Bussi's original code seems to be too simple, leading to correlated results from differnet runs.

### CODE REVIEW

**Review by hityingph — APPROVED** [review]

**Comment by brucefan1983** [comment]
In the figure below, old and new means the code before and after this PR. 1 and 2 refer to two independent runs (with different initial velocities). It can be seen that the two simulations are correlated using the old code.


image

File: src/integrate/svr_utilities.cuh [hunk 1]
```diff
@@ -83,8 +46,10 @@ static double gasdev()
   }
 }
 
-static double gamdev(const int ia)
+static double gamdev(const int ia, std::mt19937& rng)
 {
+  std::uniform_real_distribution<double> rand1(0, 1);
```
Comment (Lazemare) [inline src/integrate/svr_utilities.cuh:51]: Would a `static` declaration improve the performance?

**Comment by Lazemare** [comment]
I think the code is good enough, expect the only one question I commented on above.

File: src/integrate/svr_utilities.cuh (hunk 1 above)
Comment (brucefan1983) [reply src/integrate/svr_utilities.cuh:51]: using static variables might make the function a little bit faster, but any CPU part is not a bottleneck in GPUMD. Using static variables are not good for possible future multi-threaded extension. So I do not use static variables at all.

Comment (Lazemare) [reply src/integrate/svr_utilities.cuh:51]: ok, that makes sense :+1:

**Review by Lazemare — APPROVED** [review]

**Pull request merged**